php-checks/src/main/java/org/sonar/php/checks/phpunit/AssertTrueInsteadOfDedicatedAssertCheck.java

```diff
@@ -89,15 +95,38 @@ public void visitFunctionCall(FunctionCallTree fct) {
 
   private void checkBooleanExpressionInAssertMethod(FunctionCallTree problematicAssertionCall, String assertionName) {
     CheckUtils.argumentValue(problematicAssertionCall, "", 0).ifPresent(argumentExpression -> {
-      Optional<ReplacementAssertion> replacementAssertionOpt = getReplacementAssertion(argumentExpression);
-      if (assertionName.equals("assertFalse")) {
-        replacementAssertionOpt = replacementAssertionOpt.map(COMPLEMENTS::get);
+      Optional<String> argumentValue = retrieveBooleanLiteralFromBinaryExpression(argumentExpression);
+
+      if (argumentValue.isPresent()) {
+        // Boolean comparison, raise an issue to simplify it
+        BinaryExpressionTree bet = (BinaryExpressionTree) argumentExpression;
+        String booleanLiteralToRemove = argumentValue.get();
+        boolean shouldChangeAssertion = shouldChangeAssertionFunction(assertionName, booleanLiteralToRemove, bet);
+        String additionalAction = "";
+        if (shouldChangeAssertion) {
+          additionalAction = String.format("using %s() and ", ASSERT_TRUE.equals(assertionName) ? ASSERT_FALSE : ASSERT_TRUE);
+        }
```

Comment:
```suggestion
        String additionalAction = shouldChangeAssertionFunction(assertionName, booleanLiteralToRemove, bet) ? 
            String.format("using %s() and ", assertionName) : "";
```

Reply:
Does not work as it is, as it is missing the ternary expression `ASSERT_TRUE.equals(assertionName) ? ASSERT_FALSE : ASSERT_TRUE`.
I cannot also nest it, because it gets too complicated and even raise an issue S3358.
I adjusted with a `ASSERTIONS_COMPLEMENT` map, looks good 👍